### ISSUE TITLE
No confirmation when asking to reset Cura to factory settings.

### ISSUE THREAD

**Issue by kirilledelman** [Ultimaker/Cura#4328]
**Application Version**
3.4.0
3.4.1

**Platform**
Mac OS

**Printer**
Custom FDM Printer

**Steps to Reproduce**
At application start Cura displays a notification on the bottom that some of Cura settings are corrupt. It's a large message that covers half of the screen (lists multiple old profiles, some were deleted). The message can be dismissed by clicking X in upper right. The message also has a button "Reset". I clicked it accidentally instead of X and it wiped all my printer, carefully tuned profiles, and materials without confirmation. Hair pulling disaster event!

**Actual Results**
Cura doesn't display "Are you sure you want to delete all your custom printer profiles, print profiles, and materials?"

**Expected results**
Cura should confirm "Are you sure you want to delete all your custom printer profiles, print profiles, and materials?" before self destructing.

I was able to get this message to show up, simply by duplicating a generic PLA material:
![screen shot 2018-09-04 at 11 23 40 am](https://user-images.githubusercontent.com/6539713/45044516-f251c300-b035-11e8-90a7-4a25e623ad9b.png)

**Comment by NicholasTracy** [comment]
I can confirm that this is an issue that needs to be addressed. I just experienced this with 3.4.1. The error in question popped up after duplicating the generic material profile and changing the default temp for the duplicated profile. When clicking reset Cura actually crashed and I had to restart Cura, when it restarted I had to start over from scratch on all my configuration settings and profiles. Very frustrating.
